core/math/vector3.h

```diff
@@ -400,14 +401,23 @@ void Vector3::zero() {
 	x = y = z = 0;
 }
 
-Vector3 Vector3::slide(const Vector3 &p_vec) const {
-
-	return p_vec - *this * this->dot(p_vec);
+// slide returns the component of the vector along the given plane, specified by its normal vector.
+Vector3 Vector3::slide(const Vector3 &p_n) const {
+#ifdef DEBUG_ENABLED
```

Comment:
after this change my character barely moves in the ground because of the slide function.
I get a lot of errors coming from this
![error](https://cloud.githubusercontent.com/assets/1642015/24829381/50f68048-1c68-11e7-9cf0-3f57b25990ab.png)